src/lightning/fabric/utilities/cloud_io.py

```diff
@@ -76,7 +76,10 @@ def _atomic_save(checkpoint: Dict[str, Any], filepath: Union[str, Path]) -> None
     bytesbuffer = io.BytesIO()
     log.debug(f"Saving checkpoint: {filepath}")
     torch.save(checkpoint, bytesbuffer)
-    with fsspec.open(filepath, "wb") as f:
+
+    # We use a transaction here so that if the save gets interrupted, existing checkpoints are not trashed.
+    fs, _ = fsspec.core.url_to_fs(str(filepath))
```

Comment:
```suggestion
    fs = get_filesystem(filepath)
```

Reply:
That's a reasonable suggestion. I acutally just committed a change to do this a bit differently and use the returned urlpath in conjunction with the filesystem since I think that's a bit more idiomatic with fsspec.